pipelines/main/launch_unsigned_jobs.yml

```diff
@@ -12,6 +12,33 @@
 # and only need to touch the webui configuration when we need to alter
 # something about the privileged steps.
 
+common:
+  - diff-filter-build_plugin: &diff-filter-build
+      https://github.com/fatteneder/diff-filter-buildkite-plugin#main:
+        name: "build"
+        ignore:
+          - "*.md"
+          - "*.json"
+          - "CITATION.*"
+          - ".gitignore"
+          - ".clangd"
+          - ".mailmap"
+        # TODO: Need this option because BUILDKITE_PULL_REQUEST_BRANCH=main
+        # when running tests in Julia-CI/julia-buildkite,
+        # but for PRs against Julia/JuliaLang BUILDKITE_PULL_REQUEST_BRANCH=master
+        target_branch: "master"
+  - diff-filter-doc_plugin: &diff-filter-doc
+      https://github.com/fatteneder/diff-filter-buildkite-plugin#main:
+        name: "doc"
+        match:
+          - "doc/*"
+          - "HISTORY.md"
+          - "NEWS.md"
+        # TODO: Need this option because BUILDKITE_PULL_REQUEST_BRANCH=main
+        # when running tests in Julia-CI/julia-buildkite,
+        # but for PRs against Julia/JuliaLang BUILDKITE_PULL_REQUEST_BRANCH=master
+        target_branch: "master"
```

Comment:
Does this mean the docs run is only run for those matches? Don't we need to cover changes to docstrings and their doctests?

Reply:
Yeah. And docstrings can be in any file. I think we pretty much always need to build the docs.

Reply:
Docs build should be triggered by any of these two now:
```sh
 if [[ $${BUILDKITE_PLUGIN_DIFF_FILTER_TRIGGERED_BUILD} == 1 ||
                $${BUILDKITE_PLUGIN_DIFF_FILTER_TRIGGERED_DOC} == 1   ]]; then
```

But on that note: Do we always need at least one Julia build to also be able to build the docs? That's how I understand the instructions here https://github.com/JuliaCI/julia-buildkite/issues/243#issue-1444360578. If so, then that's not yet handled here.

---

Perhaps I should revert the `diff-filter-doc` for the moment, then just not ignore `HISTROY.md` and `NEWS.md` for `diff-filter-build`, and always build the docs when we build julia. A follow-up PR could then experiment with refinements.

Reply:
Yeah, we need at least one build of Julia in order to do the docs job.

I would say that it's okay to just do that one build (as well as the docs build) unconditionally, regardless of the diff.